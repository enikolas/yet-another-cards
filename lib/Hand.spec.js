'use strict'

const Hand = require('./Hand')
const Card = require('./Card')

describe('Hand', () => {
  it('should create an empty hand', () => {
    const hand = new Hand()

    expect(hand).toMatchSnapshot()
  })

  it('should add a new card to the hand', () => {
    const hand = new Hand()
    hand.add(new Card(Card.suits.hearts, Card.ranks[0]))

    expect(hand).toMatchSnapshot()
  })

  it('should remove a card from the hand when it is played', () => {
    const hand = new Hand()
    const card = new Card(Card.suits.hearts, Card.ranks[0])

    hand.add(card)
    hand.play(card)

    expect(hand).toMatchSnapshot()
  })

  describe('Printing', () => {
    const suits = [
      Card.suits.hearts,
      Card.suits.spades,
      Card.suits.diamonds,
      Card.suits.clubs
    ]

    let hand

    beforeEach(() => {
      hand = new Hand()

      for (let i = 0; i < suits.length; i++) {
        for (let j = 0; j < suits.length; j++) {
          hand.add(new Card(suits[i], Card.ranks[j]))
        }
      }
    })

    it('should return a readable value of the hand in a single line', () => {
      expect(hand.toString()).toMatchSnapshot()
    })

    it('should return a readable value of the hand in multiple lines', () => {
      expect(hand.multipleLines()).toMatchSnapshot()
    })
  })
})
