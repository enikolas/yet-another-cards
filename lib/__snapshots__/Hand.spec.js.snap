// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hand Printing should return a readable value of the hand in a single line 1`] = `"A♥ K♥ Q♥ J♥ A♠ K♠ Q♠ J♠ A♣ K♣ Q♣ J♣ A♦ K♦ Q♦ J♦"`;

exports[`Hand Printing should return a readable value of the hand in multiple lines 1`] = `
"♥ A K Q J
♠ A K Q J
♣ A K Q J
♦ A K Q J"
`;

exports[`Hand should add a new card to the hand 1`] = `
Hand {
  "cards": Array [
    Card {
      "rank": "A",
      "suit": "♥",
    },
  ],
}
`;

exports[`Hand should create an empty hand 1`] = `
Hand {
  "cards": Array [],
}
`;

exports[`Hand should remove a card from the hand when it is played 1`] = `
Hand {
  "cards": Array [],
}
`;
